admin/includes/functions/admin_access.php

```diff
@@ -36,7 +36,7 @@ function check_page($page, $params) {
   $result = $db->Execute($sql);
   $retVal = FALSE;
   while (!$result->EOF) {
-    if ($result->fields['main_page'] != '' && defined($result->fields['main_page']) && (constant($result->fields['main_page']) == $page || constant($result->fields['main_page']) . '.php' == $page) && $result->fields['page_params'] == $page_params) {
+    if ($result->fields['main_page'] != '' && defined($result->fields['main_page']) && (constant($result->fields['main_page']) == $page || basename(constant($result->fields['main_page']), '.php') == $page) && $result->fields['page_params'] == $page_params) {

```

Comment:
```diff
-(constant($result->fields['main_page']) == $page || constant($result->fields['main_page']) . '.php' == $page)
+(constant($result->fields['main_page']) == $page || basename(constant($result->fields['main_page']), '.php') == $page)
```

Doesn't this safely reduce even further by just removing the first half? 
eg:
```diff
-(constant($result->fields['main_page']) == $page || constant($result->fields['main_page']) . '.php' == $page)
+basename(constant($result->fields['main_page']), '.php') == $page
```

Reply:
damn straight. 

absolutely valid point.